safe_transaction_service/history/indexers/tx_processor.py

```diff
@@ -325,9 +325,15 @@ def store_new_safe_status(
     def process_decoded_transaction(
         self, internal_tx_decoded: InternalTxDecoded
     ) -> bool:
-        processed_successfully = self.__process_decoded_transaction(internal_tx_decoded)
-        internal_tx_decoded.set_processed()
-        self.clear_cache()
+        contract_address = internal_tx_decoded.internal_tx._from
+        self.clear_cache(safe_address=contract_address)
```

Comment:
Would it be enough to clear the cache in the `finally` part?

Reply:
I'm worried about one process interrupted or a task killed and not executing the finally part

Reply:
Do you think it's overkilling?

Reply:
Taking into account your previous comment, maybe it's necessary. In my first review, I found it a bit strange.